.github/workflows/tests.yml

```diff
@@ -0,0 +1,50 @@
+name: Tests
+
+on:
+  pull_request:
+  push:
+    branches: [master, main]
+
+jobs:
+  linux-tests:
+    runs-on: ubuntu-latest
+    container:
+      image: qgis/qgis:${{ matrix.qgis-image-tag }}
+    strategy:
+      matrix:
+        qgis-image-tag:
+          - release-3_22
+          # - release-3_26
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v3
+      - run: |
+          pip3 install virtualenv && virtualenv .venv --system-site-packages
+      - run: |
+          .venv/bin/pip3 install -q -r requirements.txt -r requirements-dev.txt --no-deps --only-binary=:all: && xvfb-run --server-args="+extension GLX -screen 0 1024x768x24" .venv/bin/pytest -v --cov=pickLayer --cov-report=xml test
+      # Upload coverage report. Will not work if the repo is private
+      - if: ${{ matrix.docker_tags == 'latest' && !github.event.repository.private }}
```

Comment:
this does not actually do anything, since we dont run against the latest tag. since there seems to be no easy way to set a varible for a single matrix combination inside the matrix, maybe this could be changed to match one value from the matrix, and add a comment in the matrix that the value is also used in another place (so we dont accidentally update the matrix and leave a now-removed tag here)

Reply:
I also noticed that I had wrong matrix name (docker_tags) here. I synchronized the namings.